nemoguardrails/rails/llm/llmrails.py

```diff
@@ -448,14 +448,14 @@ def _get_events_for_messages(self, messages: List[dict], state: Any):
                         }
                     )
 
-                    # If it's not the last message, we also need to add the `UserMessage` event
-                    if idx != len(messages) - 1:
-                        events.append(
-                            {
-                                "type": "UserMessage",
-                                "text": msg["content"],
-                            }
-                        )
+                    # # If it's not the last message, we also need to add the `UserMessage` event
+                    # if idx != len(messages) - 1:
+                    #     events.append(
+                    #         {
+                    #             "type": "UserMessage",
+                    #             "text": msg["content"],
+                    #         }
+                    #     )
```

Comment:
Can you add a test where it breaks without the fix? The `UserMessage` event is part of the normal flow and I don't see an obvious reason why this should be commented. Maybe there's a different root cause for the behaviour you are seeing. 

As a quick check, the test here: https://github.com/NVIDIA/NeMo-Guardrails/blob/develop/tests/test_issue_385.py works correctly. Even though the test does not check explicitly, I've just run it manually and it correctly returns the right response.